### ISSUE TITLE
Development issues/questions

### ISSUE THREAD

**Issue by snoe** [clojure-lsp/clojure-lsp#874]
Jumping back into development there's a few issues that I think are difficult and could be better.
I do not know deps.edn or kaocha well enough to make these work:

## Inability to run outside the `make` binary
Where it is painful and where I would rather launch with an equivalent  `clj -M:run`. This was working but broke at some point in the last year (component split?).

I used to launch with this command when developing (bash was required to launch in the working dir):
```
      "command": "bash",
      "args": ["-c", "cd /Users/case/dev/lsp/ && clj -M:run"],
```

### Why I want it: 
So developing with a repl to the running instance works most of the time. The difference between a lsp restart with the binary vs a lein run is that the restart will pickup saved changes. Recreating repl state in proper load order etc is not easy when juggling a number of files or instances.

In the last week, places where I've wanted this, that forced a slow `make`:
- If the process crashes I lose repl state. 
- If I want to jump between instances in different projects or editors.
- It seems that `require` of new files is broken, I believe because they are not in the binary.
- test files are not in the binary, so they cannot be loaded.

## Tests
- Running `clj -M:test --watch` does not work, I believe it should to be able to run all unit tests.
- Changing common-test does not reload watch while watch lib tests (maybe others).
- kaocha has the ability to run with `--focus-meta` which is incredibly helpful for switching between focus and broad changes. I would like to add a `test-focus` task that adds `--focus-meta :lsp/focus` and another pre-commit hook to block `:lsp/focus` to make sure they aren't being left in after dev

**Comment by ericdallo** [comment]
@snoe makes sense, I always `make`, but your points makes sense for development indeed, but I think they still work, but a little bit different, could you try `cd /Users/case/dev/lsp/cli && clj -M:run`? maybe we could have a `make run` for that too?

**Comment by ericdallo** [comment]
I usually just jack in for tests and don't use kaocha at all, only cider, but I can see how convenient for other editors that could be. Not sure how could we improve that, maybe having a test alias on `deps.edn` root which would run for all projects, WDYT?

**Comment by snoe** [comment]
> @snoe makes sense, I always `make`, but your points makes sense for development indeed, but I think they still work, but a little bit different, could you try `cd /Users/case/dev/lsp/cli && clj -M:run`? maybe we could have a `make run` for that too?

It looks like it sort of works, if I manually merge the nrepl stuff from :debug into :run alias, it's up and running. I'm not sure if there's a way to use multiple from the command line. `make run` could work, I guess just having documentation around this alternative would be good.

**Comment by ericdallo** [comment]
@snoe `cd /Users/case/dev/lsp/cli && clj -M:debug:run` should add the debug profiler

**Comment by ericdallo** [comment]
I think it's valid to add a `make run-debug`
